src/installer/installer.go

```diff
@@ -184,6 +191,42 @@ func (i *installer) InstallNode() error {
 	return i.finalize()
 }
 
+func (i *installer) waitForWorkers(ctx context.Context) error {
+	// This waiting condition is valid only for agent-based installer.
+	kc, err := i.kcBuilder(KubeconfigPath, i.log)
+	if err != nil {
+		i.log.Error(err)
+		return err
+	}
+	if invoker := common.GetInvoker(kc, i.log); invoker != common.InvokerAgent {
+		return nil
+	}
+
+	return utils.WaitForPredicate(waitForeverTimeout, 30*time.Second, func() bool {
+		workers, err := i.inventoryClient.ListsHostsForRole(ctx, string(models.HostRoleWorker))
+		if err != nil {
+			i.log.WithError(err).Error("failed to get workers")
+			return false
+		}
+		if len(workers) == 0 {
+			return true
+		}
+
+		i.log.Infof("Found %d workers, waiting for workers to reach reboot stage", len(workers))
+		for _, worker := range workers {
+			// If any worker is still in the waiting for control plane stage, then
+			// it means it didn't reboot yet and still requires assisted-service
+			if worker.Progress.CurrentStage == models.HostStageWaitingForControlPlane {
```

Comment:
The worker might have an earlier stage (e.g. HostStageInstalling in case of a very slow disk)
So you might want to change this: 
funk.Contains([]models.HostStage{models.HostStageWaitingForControlPlane, models.HostStageInstalling, models.HostStageStartingInstallation) c.host.Progress.CurrentStage)